pidgin/app.py

```diff
@@ -20,21 +31,25 @@ def get_json_metadata(object_id):
         return e.message, e.code
 
 
-@app.route('/bibtex/<path:object_id>')
 def get_bibtex_metadata(object_id):
     """
     Get core metadata as BibTeX from an object_id.
     """
-    metadata = get_metadata_dict(object_id)
-    return translate_dict_to_bibtex(metadata)
+    try:
+        metadata = get_metadata_dict(object_id)
+        return translate_dict_to_bibtex(metadata)
+    except PidginException as e:
+        return e.message, e.code
 
 
 def get_metadata_dict(object_id):
     """
     Create a dictionary containing the metadata for a given object_id.
     """
     response = request_metadata(object_id) # query to peregrine
-    return flatten_dict(response) # translate the response to a dictionary
+    dict = flatten_dict(response)
```

Comment:
As a general rule avoid using built-ins and standard types (like `dict`) for variable names, since this overwrites the built-in (and there's usually a more informative name to use anyways). So in this case, `metadata` maybe?